### PR TITLE
Sync with upstream vitess; 2018-12-14

### PR DESCRIPTION
**Pull request by setassociative**
tl;dr: all changes look :+1:

## Slack changes
-  Adds removed vschema  
(for examples)  
7f0ee4c74 Merge pull request #4434 from tinyspeck/add-vschema-local-examples

- Stops reusing a consul lock in master election  
8e64c77c0 (tag: v3.0.0-rc.1) Merge pull request #4353 from tinyspeck/fix-for-consul-master-election

- Adds support for vschema alter syntax  
eccf91289 Merge pull request #4412 from tinyspeck/vschema-alter-syntax

- vtctl support for vschema ddls  
3b2294b7e Merge pull request #4385 from tinyspeck/vtctl-support-vschema-ddls

- remove NFS folder syncing as default in vagrant config  
fb8c46f78 Merge pull request #4424 from tinyspeck/brirams/remote_nfs_from_vagrantfile

## Non-Slack changes
- java: Bump SNAPSHOT version to 3.1.0-SNAPSHOT after Vitess release 3.0  
safe because it's java stuff  
eb5a5e0e4 (vupstream/master, vup_mas) Merge pull request #4453 from mpawliszyn/mikepaw.update-snapshot

- Parents the parent pom of the standard sonatype oss pom for better defaults  
safe, more java stuff  
84a352400 Merge pull request #4447 from mpawliszyn/mikepaw.better-maven-defaults

- Allow subclasses of GrpcClientFactory to modify channel builder  
safe bc java grpc client work  
be9745195 Merge pull request #4439 from smourachov/grpc-java-allow-modify-channel-builder

- local example: fix typos  
safe: examples  
4f192d100 (tag: v3.0.0-rc.3, tag: v3.0) Merge pull request #4438 from planetscale/ss-local-fix

- helm: add missing default.
safe helm support  
43aa7e934 (tag: v3.0.0-rc.2) Merge pull request #4437 from planetscale/ss-default

- Removes secure_file_priv and innodb_support_xa from master mysql 56 cnf  
safe, config change used in examples  
8c567f765 Merge pull request #4419 from derekperkins/remove-mycnf

- Add scope to oauth token for gcs backup storage client  
safe, updates auth flag for GCS based backup storage  
ecb972f2a Merge pull request #4433 from Santiclause/gcs-backup-scope

- Enhance local example  
safe, example code  
2a950238d Merge pull request #4429 from planetscale/ds-add-to-example

- WIP: adapt helm/k8s example to run locally  
safe, updates helm/k8s example  
f2118cc41 Merge pull request #4427 from planetscale/ds-local-example

## Source material

```
$ git log tinyspeck/master..upstream/master --merges --oneline
eb5a5e0e4 (upstream/master) Merge pull request #4453 from mpawliszyn/mikepaw.update-snapshot
84a352400 Merge pull request #4447 from mpawliszyn/mikepaw.better-maven-defaults
be9745195 Merge pull request #4439 from smourachov/grpc-java-allow-modify-channel-builder
4f192d100 (tag: v3.0.0-rc.3, tag: v3.0) Merge pull request #4438 from planetscale/ss-local-fix
7f0ee4c74 Merge pull request #4434 from tinyspeck/add-vschema-local-examples
43aa7e934 (tag: v3.0.0-rc.2) Merge pull request #4437 from planetscale/ss-default
8c567f765 Merge pull request #4419 from derekperkins/remove-mycnf
ecb972f2a Merge pull request #4433 from Santiclause/gcs-backup-scope
8e64c77c0 (tag: v3.0.0-rc.1) Merge pull request #4353 from tinyspeck/fix-for-consul-master-election
eccf91289 Merge pull request #4412 from tinyspeck/vschema-alter-syntax
2a950238d Merge pull request #4429 from planetscale/ds-add-to-example
f2118cc41 Merge pull request #4427 from planetscale/ds-local-example
3b2294b7e Merge pull request #4385 from tinyspeck/vtctl-support-vschema-ddls
fb8c46f78 Merge pull request #4424 from tinyspeck/brirams/remote_nfs_from_vagrantfile
```

### CODE REVIEW

**Comment by rafael** [comment]
LGTM +1
